BrightFutures/AsyncType+Debug.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AsyncType+Debug.swift
+//  BrightFutures
+//
+//  Created by Oleksii on 23/09/2016.
+//  Copyright © 2016 Thomas Visser. All rights reserved.
+//
+
+import Result
+
+public protocol LoggerType {
+    func log(message: String)
+}
+
+fileprivate struct Logger: LoggerType {
+    func log(message: String) {
+        print(message)
+    }
+}
+
+public extension AsyncType where Value: ResultProtocol {
+    public func debug(logger: LoggerType = Logger(), context c: @escaping ExecutionContext = DefaultThreadingModel(), messageBlock: @escaping (Self.Value) -> String) -> Self {
+        return andThen(context: c, callback: { result in
+            logger.log(message: messageBlock(result))
+        })
+    }
+    
+    public func debug(_ identifer: String? = nil, logger: LoggerType = Logger(), file: String = #file, line: UInt = #line, function: String = #function, context c: @escaping ExecutionContext = DefaultThreadingModel()) -> Self {
+        return debug(logger: logger, context: c, messageBlock: { result -> String in
+            let messageBody: String
```

Comment:
So do you want to get rid of the first debug method that takes closure as param at all?